feat: Add apps proxy local development docs

Matovidlo opened the pull request:
Jira: [PSGO-919](https://keboola.atlassian.net/browse/PSGO-919)

**Changes:**
- Add apps proxy documentation for local development.
- Include short overview of apps proxy, what is it, why we have it and how to use it.
- Add basic configuration of the apps proxy for development on local machine.

-----------


[PSGO-919]: https://keboola.atlassian.net/browse/PSGO-919?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jachym-tousek-keboola commented on docs/apps-proxy/overview.md:
```diff
@@ -0,0 +1,68 @@
+# Apps proxy Architecture Overview
+
+- Serves for data apps authentication and authorization.
+- Typicall usage is to perform OIDC login through some OIDC provider (e.g Microsoft login, google login etc.)
```
```suggestion
- Typicall usage is to perform OIDC login through some OIDC provider (e.g Microsoft login, Google login etc.)
```

Matovidlo replied:
Fixed

jachym-tousek-keboola commented on docs/apps-proxy/overview.md:
````diff
@@ -0,0 +1,68 @@
+In project directory run:
+
+```
+docker compose run --rm --service-ports dev --net=my-test bash
````
This doesn't work for me:

```
Error response from daemon: failed to create task for container: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "--net=my-test": executable file not found in $PATH: unknown
```

I'm guessing some additional setup is necessary?

Matovidlo replied:
Reworked using `network` in docker compose. It is more reliable.

jachym-tousek-keboola commented on docs/apps-proxy/overview.md:
````diff
@@ -0,0 +1,68 @@
+Inside this bash run:
+
+```
+make run-app-proxy
````
This doesn't work either.

```
make: *** No rule to make target 'run-app-proxy'.  Stop.
```

Looks like a typo.

```suggestion
make run-apps-proxy
```

jachym-tousek-keboola replied:
However `make run-apps-proxy` doesn't work either:

```
make[1]: Entering directory '/code'
CGO_ENABLED=0 go build -v -mod mod -ldflags "-s -w" -o "./target/apps-proxy/proxy" ./cmd/apps-proxy
# cd /code; git status --porcelain
fatal: detected dubious ownership in repository at '/code'
To add an exception for this directory, call:

        git config --global --add safe.directory /code
error obtaining VCS status: exit status 128
        Use -buildvcs=false to disable VCS stamping.
make[1]: *** [Makefile:52: build-apps-proxy] Error 1
make[1]: Leaving directory '/code'
```

Matovidlo replied:
Fixed. I found that it is more beneficial to uncomment `command` section in docker compose and run it using `docker compose up dev`

jachym-tousek-keboola commented on docs/apps-proxy/overview.md:
````diff
@@ -0,0 +1,68 @@
+```
+docker run --net=cli_default --rm \
+  --env DOMAIN=test.hub.keboola.local \
+  --env TARGET_HOST=<containerid> \
````
Remove this extra step to obtain container id. It should be possible to just replace `<containerid>` with this:

```
$(docker ps -aqf "name=dev")
```

Matovidlo replied:
Thanks for advice. I went for docker compose instead. Check last commit

jachym-tousek-keboola reviewed (changes requested):
Good work, I just want the containerid thing automated.

Matovidlo commented:
@jachym-tousek-keboola I have changed the docker mixed setup with docker compose to only docker compose. There is only one adjustement needed and it is uncommenting of `command` step in docker compose. Otherwise rest of workflow should work.
I adjusted provisioning of dev as it was not possible to override options by the `environment` of `docker compose` `dev` container.

jachym-tousek-keboola commented on docs/apps-proxy/overview.md:
````diff
@@ -0,0 +1,52 @@
+In project directory uncommect in [docker-compose.yml](../../docker-compose.yml) the `command` section and fill it with `apps-proxy` service. It should look like this
+```
+command: >
+    sh -c "git config --global --add safe.directory /code
+           make run-apps-proxy"
````
I don't think the git config should be here, I think that needs to be run outside of the container manually.

Matovidlo replied:
You cannot run it in `command` section without this. I am not sure why it is required as it was not required before. Any suggestions how to fix this?

jachym-tousek-keboola replied:
Apparently git checks if the user running the git command is the owner of the repository and throws "dubious ownership" error. I checked online and this seems to be the recommended fix. Okay then.
